buffer/src/main/java/io/netty/buffer/AbstractPooledDerivedByteBuf.java

```diff
@@ -269,15 +269,13 @@ boolean release0(int decrement) {
 
         @Override
         public ByteBuf duplicate() {
-            // Capacity is not allowed to change for a sliced ByteBuf, so length == capacity()
-            final ByteBuf duplicate = slice(0, capacity());
-            duplicate.setIndex(readerIndex(), writerIndex());
-            return duplicate;
+            return new PooledNonRetainedDuplicateByteBuf(referenceCountDelegate, unwrap())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Prefix this call to "unwrap" with "super.". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2388)